ioengines.c

```diff
@@ -311,6 +313,38 @@ enum fio_q_status td_io_queue(struct thread_data *td, struct io_u *io_u)
 					sizeof(io_u->issue_time));
 	}
 
+	if (fio_option_is_set(o, ioprio) ||
```

Comment:
I'm not convinced this part is even workable. In practice, per-io priority should be passed in during submission, through libaio actually utilizing the priority fields. That would be a lot more straight forward, and also a lot faster. This setup relies on the fact that we don't look at the task io priority after the IO has been queued in the kernel, It seems fragile.

Reply:
I was hoping to make the prio_percent option workable on all IO engines rather than just libaio. Do you know if other IO engines can directly set the priority fields? Would it make sense to add a new function to the ioengine_ops struct called prio_prep and create a libaio specific prio_prep that sets the priority fields directly but leave other IO engines to use this current suboptimal implementation until they are updated with an engine specific implementation?